core/workspace_comment_render_svg.js

```diff
@@ -276,40 +368,46 @@ Blockly.WorkspaceCommentSvg.prototype.resizeMouseDown_ = function(e) {
   e.stopPropagation();
 };
 
-/**
- * Handle a mouse-down on comment's delete icon.
- * @param {!Event} e Mouse down event.
- * @private
- */
-Blockly.WorkspaceCommentSvg.prototype.deleteMouseDown_ = function(e) {
-  // highlight the delete icon
-  Blockly.utils.addClass(
-      /** @type {!Element} */ (this.deleteIconBorder_), 'blocklyDeleteIconHighlighted');
-  // This event has been handled.  No need to bubble up to the document.
-  e.stopPropagation();
-};
 
 /**
- * Handle a mouse-out on comment's delete icon.
- * @param {!Event} e Mouse out event.
+ * Set the minimized state of the bubble.
+ * @param {boolean} minimize Whether the bubble should be minimized
+ * @param {?string} labelText Optional label text for the comment top bar
+ *    when it is minimized.
  * @private
  */
-Blockly.WorkspaceCommentSvg.prototype.deleteMouseOut_ = function(/*e*/) {
-  // restore highlight on the delete icon
-  Blockly.utils.removeClass(
-      /** @type {!Element} */ (this.deleteIconBorder_), 'blocklyDeleteIconHighlighted');
-};
-
-/**
- * Handle a mouse-up on comment's delete icon.
- * @param {!Event} e Mouse up event.
- * @private
- */
-Blockly.WorkspaceCommentSvg.prototype.deleteMouseUp_ = function(e) {
-  // Delete this comment
-  this.dispose(true, true);
-  // This event has been handled.  No need to bubble up to the document.
-  e.stopPropagation();
+Blockly.WorkspaceCommentSvg.prototype.setRenderedMinimizeState_ = function(minimize, labelText) {
+  if (minimize) {
+    // Change minimize icon
+    this.minimizeArrow_.setAttributeNS('http://www.w3.org/1999/xlink',
+        'xlink:href', Blockly.mainWorkspace.options.pathToMedia + 'comment-arrow-up.svg');
+    // Hide text area
+    this.commentEditor_.setAttribute('display', 'none');
+    // Hide resize handle if it exists
+    if (this.resizeGroup_) {
+      this.resizeGroup_.setAttribute('display', 'none');
+    }
+    if (labelText && this.labelText_ != labelText) {
+      // Update label and display
+      // TODO is there a better way to do this?
```

Comment:
https://github.com/LLK/scratch-blocks/commit/02da1b69712356210c7ff3e6a769e60a7880dde1

Reply:
Thanks for the pointer! Done (here and in ScratchBubble).